### PR TITLE
chore(downloader): simplify bodies task polling

### PR DESCRIPTION
**Pull request by rkrasiuk**
## Description

Simplify bodies task downloader polling logic.

### CODE REVIEW

File: crates/net/downloaders/src/bodies/task.rs [hunk 1]
```diff
@@ -115,35 +115,23 @@ impl<T: BodyDownloader> Future for SpawnedDownloader<T> {
...
-                                    // we can also exit
-                                    return Poll::Ready(())
-                                }
-                            }
+            while let Poll::Ready(update) = this.updates.poll_next_unpin(cx) {
+                if let Some(range) = update {
+                    if let Err(err) = this.downloader.set_download_range(range) {
+                        tracing::error!(target: "downloaders::bodies", ?err, "Failed to set bodies download range");
+
+                        let forward_error_result = ready!(this.bodies_tx.poll_reserve(cx))
```
Comment (mattsse) [inline crates/net/downloaders/src/bodies/task.rs:123]: even though this should not really be reachable, this still throws away the error if the channel is currently full.

we could clone the `bodies_tx` first which would ensure a new slot for the error

**Review by mattsse — CHANGES_REQUESTED** [review]

File: crates/net/downloaders/src/bodies/task.rs [hunk 2]
```diff
@@ -120,9 +120,9 @@ impl<T: BodyDownloader> Future for SpawnedDownloader<T> {
                     if let Err(err) = this.downloader.set_download_range(range) {
                         tracing::error!(target: "downloaders::bodies", ?err, "Failed to set bodies download range");
 
-                        let forward_error_result = ready!(this.bodies_tx.poll_reserve(cx))
-                            .and_then(|_| this.bodies_tx.send_item(Err(err)));
-                        if forward_error_result.is_err() {
+                        // Clone the sender ensure its availability. See [PollSender::clone].
+                        let mut bodies_tx = this.bodies_tx.clone();
+                        if bodies_tx.send_item(Err(err)).is_err() {
```
Comment (mattsse) [inline crates/net/downloaders/src/bodies/task.rs:125]: this will panic, still needs a poll_reserve call

**Review by mattsse — CHANGES_REQUESTED** [review]

File: crates/net/downloaders/src/bodies/task.rs (hunk 2 above)
Comment (rkrasiuk) [reply crates/net/downloaders/src/bodies/task.rs:125]: even though we know for sure it's either Idle or Closed? interesting...

**Review by mattsse — APPROVED** [review]
lgtm

**Pull request merged**